### DISCUSSION TITLE
scheduledTimer

### DISCUSSION THREAD

**Discussion by AndreasWagnerBerlin** [sstaub/Ticker#35] [Ideas]
Dear Stefan,
how about the idea to extend the Timer Library with the  feature of a scheduled Timer.
I think of a normal timer with an additional parameter indicating the start time.
This would make it possible to start different timers in a shifted manner.

If this is implementable with existing features, I would be interested in knowing more details.

Regards Andreas

**Comment by sstaub** [comment]
Did you think of a start delay? You can try a one shot Ticker which starts another Ticker.

**Comment by AndreasWagnerBerlin** [comment]
Yes, I thought of a start delay. However, your proposal works when I put the start as follows:

``` 
Ticker timer1(startMyTimer, 1500, 1);
...
void startMyTimer() {
    timer3.start();
}
...
timer1.start();
```

**Reply by AndreasWagnerBerlin** [reply]
Another idea would be to enable Ticker to allow "timer3.start() as callback function.
`Ticker timer1 (timer3.start(), 1500, 1);`

**Reply by sstaub** [reply]
Did you tried it, which board?

**Reply by AndreasWagnerBerlin** [reply]
I tried using M5Stack Core 2, i.e ESP32.
The IDE is Microsoft Visual Studio Community, latest version.
The following two screenshots show the error messages:
![2021-03-11 10_48_49-M5Ticker - Microsoft Visual Studio](https://user-images.githubusercontent.com/8362970/110769143-2b4afe00-8258-11eb-83e6-ca2367716896.gif)

![2021-03-11 10_50_38-Microsoft Visual Studio](https://user-images.githubusercontent.com/8362970/110769160-2f771b80-8258-11eb-96ca-f19d46c5ab1b.gif)

**Reply by sstaub** [reply]
Ok, I hoped this would work with functional, you must use the workaround.
